sync/awsdatabricks.py

```diff
@@ -388,10 +381,69 @@ def write_file(body: bytes):
                     "Reservations": [*removed_instances, *updated_instances, *new_instances]
                 }
 
-            write_file(orjson.dumps(instances))
+            # Now record the ebs volumes
+            volumes = _get_ebs_volumes(cluster_id, ec2)
+
+            for v in volumes:
+                if v["VolumeId"] not in recorded_volume_id_set:
+                    recorded_volume_id_set.add(v["VolumeId"])
+                    recorded_volumes.append(v)
 
+            instances["Volumes"] = recorded_volumes
+
+            write_file(orjson.dumps(instances))
             previous_instances = instances
         except Exception as e:
             logger.error(f"Exception encountered while polling cluster: {e}")
 
         sleep(polling_period)
+
+
+def _get_ebs_volumes(cluster_id: str, ec2_client: "botocore.client.ec2") -> list[dict]:
+
+    args = {}
+    volumes = []
+    while True:
+        response = ec2_client.describe_volumes(
+            Filters=[
+                {"Name": "tag:Vendor", "Values": ["Databricks"]},
+                {"Name": "tag:ClusterId", "Values": [cluster_id]},
+            ],
+            **args,
+        )
+
+        if new_volumes := response.get("Volumes"):
+            volumes += new_volumes
+
+        if next_token := response.get("NextToken"):
+            args["NextToken"] = next_token
+        else:
+            break
+
+    return volumes
+
+
+def _get_ec2_instances(cluster_id: str, ec2_client: "botocore.client.ec2") -> dict[str, list]:
+
+    args = {}
+    instances = []
+    while True:
+        response = ec2_client.describe_instances(
+            Filters=[
+                {"Name": "tag:Vendor", "Values": ["Databricks"]},
+                {"Name": "tag:ClusterId", "Values": [cluster_id]},
+            ],
+            **args,
+        )
+
+        print(response)
+
+        if new_instances := response.get("Reservations"):
+            instances += new_instances
+
+        if next_token := response.get("NextToken"):
+            args["NextToken"] = next_token
+        else:
+            break
+
+    return {"Reservations": instances}
```

Comment:
The `describe_volumes` call is new. I also factored out the `describe_instances` call because it was common to two places, and it was a good opportunity to beef it up with some NextToken handling for very large clusters.